src/main/java/frc/robot/Robot.java

```diff
@@ -247,12 +306,123 @@ public void robotPeriodic() {
+            })
+        .andThen(
+            () -> {
+              AutoAimStates.startingAutonomousSeconds = Timer.getFPGATimestamp();
+            },
+            swerve);
+  }
+
+  /**
+   * A demo command that goes through all the steps of a shoot while moving algorithm Has print
+   * commands for any unimplemented functionality
+   *
+   * @param speeds
+   * @return A command that takes the robot through an auto aim sequence
+   */
+  public Command teleopAutoAim(Supplier<ChassisSpeeds> speeds) {
+    Command getInitialValues =
+        Commands.runOnce(
+            () -> {
+              AutoAimStates.curShotSpeeds = speeds.get();
+              Logger.recordOutput("AutoAim/cur shot speedd", AutoAimStates.curShotSpeeds);
+              AutoAimStates.curShotData =
+                  AutoAim.shotMap.get(
+                      swerve
+                          .getLinearFuturePose(
+                              AutoAim.LOOKAHEAD_TIME_SECONDS, AutoAimStates.curShotSpeeds)
+                          .minus(FieldConstants.getSpeaker())
+                          .getTranslation()
+                          .getNorm());
+              System.out.println(Timer.getFPGATimestamp());
+            });
+    Command runRobot =
+        Commands.parallel(
+            shooter.runStateCmd(
+                () -> AutoAimStates.curShotData.getRotation(),
+                () -> AutoAimStates.curShotData.getLeftRPS(),
+                () -> AutoAimStates.curShotData.getRightRPS()),
+            swerve.teleopPointTowardsTranslationCmd(
+                () -> AutoAimStates.curShotSpeeds.vxMetersPerSecond,
+                () -> AutoAimStates.curShotSpeeds.vyMetersPerSecond,
+                AutoAim.LOOKAHEAD_TIME_SECONDS));
+    return Commands.sequence(
+            getInitialValues,
+            Commands.deadline(Commands.waitSeconds(AutoAim.LOOKAHEAD_TIME_SECONDS), runRobot)
+                .finallyDo(
+                    () -> {
+                      System.out.println("Shoot note");
+                    }),
+            // keeps moving to prevent the robot from stopping and changing the velocity of the note
+            swerve
+                .runVelocityFieldRelative(
+                    () ->
+                        new ChassisSpeeds(
+                            AutoAimStates.curShotSpeeds.vxMetersPerSecond,
+                            AutoAimStates.curShotSpeeds.vyMetersPerSecond,
+                            0))
+                .withTimeout(0.25))
+        .andThen(() -> System.out.println(Timer.getFPGATimestamp()), swerve);
+  }
+
+  public Command autonomousAutoAim(String pathName) {
+
+    return Commands.sequence(
+        Commands.deadline(
+                Commands.waitSeconds(AutoAim.LOOKAHEAD_TIME_SECONDS),
+                Commands.parallel(
+                    shooter.runStateCmd(
+                        () -> AutoAimStates.curShotData.getRotation(),
+                        () -> AutoAimStates.curShotData.getLeftRPS(),
```

Comment:
`AutoAimStates.curShotData::getLeftRPS` rather than the lambda form. Right and Rotation too.